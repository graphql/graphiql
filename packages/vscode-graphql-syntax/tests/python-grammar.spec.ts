import { tokenizeFile } from './__utilities__/utilities';

describe('inline.graphql.python grammar', () => {
  const scope = 'inline.graphql.python';

  it('should tokenize a simple python file', async () => {
    const result = await tokenizeFile('__fixtures__/test.py', scope);
    expect(result).toMatchSnapshot();
  });
});
