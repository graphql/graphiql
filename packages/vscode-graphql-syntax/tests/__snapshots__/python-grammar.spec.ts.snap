// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`inline.graphql.python grammar > should tokenize a simple python file 1`] = `
query =       | 
              | 
gql           | entity.name.function
(             | 
              | meta.embedded.block.graphql
"""           | meta.embedded.block.graphql string.quoted.multi.python
              | meta.embedded.block.graphql
query         | meta.embedded.block.graphql keyword.operation.graphql
              | meta.embedded.block.graphql
getContinents | meta.embedded.block.graphql entity.name.function.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
{             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
continents    | meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
{             | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
code          | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
name          | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
}             | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
}             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
"""           | meta.embedded.block.graphql string.quoted.multi.python
)             | 
              | 
query =       | 
              | 
gql           | entity.name.function
(             | 
'''           | string.quoted.multi.python
              | meta.embedded.block.graphql
query         | meta.embedded.block.graphql keyword.operation.graphql
              | meta.embedded.block.graphql
getContinents | meta.embedded.block.graphql entity.name.function.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
{             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
continents    | meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
{             | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
code          | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
name          | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
}             | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
}             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
'''           | string.quoted.multi.python
)             | 
              | 
query =       | 
              | 
gql           | entity.name.function
(             | 
'''           | meta.embedded.block.graphql string.quoted.multi.python
              | meta.embedded.block.graphql
query         | meta.embedded.block.graphql keyword.operation.graphql
              | meta.embedded.block.graphql
getContinents | meta.embedded.block.graphql entity.name.function.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
{             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
continents    | meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
{             | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
code          | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
name          | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
}             | meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
}             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
'''           | meta.embedded.block.graphql string.quoted.multi.python
)             | 
              | 
'''           | string.quoted.multi.python
#graphql      | comment.line.graphql.js
              | 
query         | keyword.operation.graphql
              | 
getContinents | entity.name.function.graphql
              | meta.selectionset.graphql
{             | meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql
continents    | meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
{             | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
code          | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
name          | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
}             | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql
}             | meta.selectionset.graphql punctuation.operation.graphql
'''           | string.quoted.multi.python
              | 
query =       | 
'''           | string.quoted.multi.python
#graphql      | comment.line.graphql.js
              | 
query         | keyword.operation.graphql
              | 
getContinents | entity.name.function.graphql
              | meta.selectionset.graphql
{             | meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql
continents    | meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
{             | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
code          | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
name          | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
}             | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
}             | meta.selectionset.graphql punctuation.operation.graphql
'''           | string.quoted.multi.python
              | 
"""           | string.quoted.multi.python
#graphql      | comment.line.graphql.js
              | 
query         | keyword.operation.graphql
              | 
getContinents | entity.name.function.graphql
              | meta.selectionset.graphql
{             | meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql
continents    | meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
{             | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
code          | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
name          | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
}             | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql
}             | meta.selectionset.graphql punctuation.operation.graphql
"""           | string.quoted.multi.python
              | 
"""           | string.quoted.multi.python
#graphql      | comment.line.graphql.js
              | 
query         | keyword.operation.graphql
              | 
getContinents | entity.name.function.graphql
              | meta.selectionset.graphql
{             | meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql
continents    | meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
{             | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
code          | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
name          | meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql meta.selectionset.graphql
}             | meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
}             | meta.selectionset.graphql punctuation.operation.graphql
"""           | string.quoted.multi.python
              | 
`;
